src/app/modules/group/components/member-list/member-list.component.ts

```diff
@@ -243,4 +258,85 @@ export class MemberListComponent implements OnChanges, OnDestroy {
         }
       });
   }
+
+  onRemoveGroup(event: Event, selectedIds: string[]): void {
+    this.confirmationService.confirm({
+      target: event.target || undefined,
+      key: 'commonPopup',
+      icon: 'pi pi-exclamation-triangle',
+      message: $localize`Are you sure you want to permanently delete ${getSelectedGroupChildCaptions(this.selection as GroupChild[])}?
+       This operation cannot be undone.`,
+      acceptLabel: $localize`Yes`,
+      acceptIcon: 'fa fa-check',
+      rejectLabel: $localize`No`,
+      accept: () => this.removeGroupsOrSubgroups(selectedIds),
+    });
+  }
+
+  onRemoveSubgroups(event: Event, groupId: string, selectedIds: string[]): void {
+    this.confirmationService.confirm({
+      target: event.target || undefined,
+      key: 'commonPopup',
+      icon: 'pi pi-exclamation-triangle',
+      message: $localize`By removing ${getSelectedGroupChildCaptions(this.selection as GroupChild[])} from the group, you may loose
+       manager access to them (if no explicit permission or through other parent group). Are you sure you want to proceed?`,
+      acceptLabel: $localize`Yes`,
+      acceptIcon: 'fa fa-check',
+      rejectLabel: $localize`No`,
+      accept: () => this.removeGroupsOrSubgroups(selectedIds, groupId),
+    });
+  }
+
+  removeGroupsOrSubgroups(selectedIds: string[], groupId?: string): void {
+    this.removalInProgress$.next(true);
+    const request$ = groupId ?
+      this.removeSubgroupService.removeBatch(groupId, selectedIds) : this.removeGroupService.removeBatch(selectedIds);
+
+    request$.subscribe({
+      next: response => {
+        displayGroupRemovalResponseToast(this.actionFeedbackService, response);
+        this.table?.clear();
+        this.unselectAll();
+        this.fetchData();
+        this.removalInProgress$.next(false);
+      },
+      error: _err => {
+        this.removalInProgress$.next(false);
+        this.actionFeedbackService.unexpectedError();
+      }
+    });
+  }
+
+  onRemove(event: Event): void {
+    if (this.selection.length === 0 || !this.groupData) return;
+
+    const groupId = this.groupData.group.id;
+    const selectedIds = this.selection.map(member => member.id);
```

Comment:
In the mapper function, it's not always a member, unless it is moved into the if statement below, it can also be a group child
I think the various use cases should be explicit, either we are dealing with `memberIds` or with `groupChildIds`, but `selectedIds` is dark

Reply:
Probably it could be covered by my fix for prev comment, please check and let me know what you are think :)